dev/tasks/linux-packages/yum/centos-8/Dockerfile

```diff
@@ -0,0 +1,64 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+ARG FROM=centos:8
+FROM ${FROM}
+
+COPY qemu-* /usr/bin/
+
+ARG DEBUG
+
+RUN \
+  quiet=$([ "${DEBUG}" = "yes" ] || echo "--quiet") && \
+  dnf install -y ${quiet} epel-release && \
+  dnf install --enablerepo=PowerTools -y ${quiet} \
+    bison \
+    boost-devel \
+    brotli-devel \
+    bzip2-devel \
+    clang \
+    cmake \
+    # double-conversion-devel \
+    flex \
+    gcc-c++ \
+    gflags-devel \
+    git \
+    glog-devel \
+    gobject-introspection-devel \
+    gtk-doc \
+    libzstd-devel \
+    llvm-devel \
+    llvm-static \
+    lz4-devel \
+    make \
+    ncurses-devel \
+    ninja-build \
+    openssl-devel \
+    pkg-config \
+    python3 \
+    python3-devel \
+    python3-numpy \
+    python3-pip \
+    re2-devel \
+    # rapidjson-devel \
+    rpm-build \
+    rpmdevtools \
+    snappy-devel \
+    tar \
+    # thrift-devel \
+    zlib-devel && \
+  dnf clean ${quiet} all
```

Comment:
Could we have a COMMAND entrypoint to build automatically? I'd like to test locally without crossbow.

Reply:
Umm, it may be difficult. We need to some preparations before running Docker container to build packages. They are done by `rake` (`make` tool in Ruby) for now.

We can test locally by the following command lines:

```console
$ cd dev/tasks/linux-packages
$ rake yum:build YUM_TARGETS=centos-8
```

Does it satisfy you?

Reply:
Once it's build, how do I access the rpm?

Reply:
There are `*.rpm` under `dev/tasks/linux-packages/yum/repositories/`:

```console
$ ls -lat yum/repositories/**/*.rpm | head -20
-rw-r--r-- 1 root root   107928 Oct 23 14:03 yum/repositories/centos/6/x86_64/Packages/plasma-devel-1.0.0-0.dev20191023.el6.x86_64.rpm
-rw-r--r-- 1 root root   388848 Oct 23 14:03 yum/repositories/centos/6/x86_64/Packages/plasma-store-server-1.0.0-0.dev20191023.el6.x86_64.rpm
-rw-r--r-- 1 root root   108824 Oct 23 14:03 yum/repositories/centos/6/x86_64/Packages/plasma-libs-1.0.0-0.dev20191023.el6.x86_64.rpm
-rw-r--r-- 1 root root   372560 Oct 23 14:03 yum/repositories/centos/6/x86_64/Packages/arrow-python-devel-1.0.0-0.dev20191023.el6.x86_64.rpm
-rw-r--r-- 1 root root   327232 Oct 23 14:03 yum/repositories/centos/6/x86_64/Packages/arrow-python-libs-1.0.0-0.dev20191023.el6.x86_64.rpm
-rw-r--r-- 1 root root   167080 Oct 23 14:03 yum/repositories/centos/6/x86_64/Packages/arrow-dataset-devel-1.0.0-0.dev20191023.el6.x86_64.rpm
-rw-r--r-- 1 root root   148932 Oct 23 14:03 yum/repositories/centos/6/x86_64/Packages/arrow-dataset-libs-1.0.0-0.dev20191023.el6.x86_64.rpm
-rw-r--r-- 1 root root  2702516 Oct 23 14:03 yum/repositories/centos/6/x86_64/Packages/arrow-devel-1.0.0-0.dev20191023.el6.x86_64.rpm
-rw-r--r-- 1 root root  2998140 Oct 23 14:03 yum/repositories/centos/6/x86_64/Packages/arrow-libs-1.0.0-0.dev20191023.el6.x86_64.rpm
-rw-r--r-- 1 root root 40473012 Oct 23 14:03 yum/repositories/centos/6/x86_64/Packages/arrow-debuginfo-1.0.0-0.dev20191023.el6.x86_64.rpm
-rw-r--r-- 1 root root  5043881 Oct 23 14:03 yum/repositories/centos/6/source/SRPMS/arrow-1.0.0-0.dev20191023.el6.src.rpm
-rw-r--r-- 1 root root    32436 Oct 23 12:05 yum/repositories/centos/7/x86_64/Packages/parquet-glib-doc-1.0.0-0.dev20191023.el7.x86_64.rpm
-rw-r--r-- 1 root root    30008 Oct 23 12:05 yum/repositories/centos/7/x86_64/Packages/parquet-glib-devel-1.0.0-0.dev20191023.el7.x86_64.rpm
-rw-r--r-- 1 root root    31424 Oct 23 12:05 yum/repositories/centos/7/x86_64/Packages/parquet-glib-libs-1.0.0-0.dev20191023.el7.x86_64.rpm
-rw-r--r-- 1 root root    34676 Oct 23 12:05 yum/repositories/centos/7/x86_64/Packages/plasma-glib-doc-1.0.0-0.dev20191023.el7.x86_64.rpm
-rw-r--r-- 1 root root    36744 Oct 23 12:05 yum/repositories/centos/7/x86_64/Packages/plasma-glib-devel-1.0.0-0.dev20191023.el7.x86_64.rpm
-rw-r--r-- 1 root root    36988 Oct 23 12:05 yum/repositories/centos/7/x86_64/Packages/plasma-glib-libs-1.0.0-0.dev20191023.el7.x86_64.rpm
-rw-r--r-- 1 root root   149284 Oct 23 12:05 yum/repositories/centos/7/x86_64/Packages/arrow-glib-doc-1.0.0-0.dev20191023.el7.x86_64.rpm
-rw-r--r-- 1 root root   260476 Oct 23 12:05 yum/repositories/centos/7/x86_64/Packages/arrow-glib-devel-1.0.0-0.dev20191023.el7.x86_64.rpm
-rw-r--r-- 1 root root   221328 Oct 23 12:05 yum/repositories/centos/7/x86_64/Packages/arrow-glib-libs-1.0.0-0.dev20191023.el7.x86_64.rpm
```